src/gc/gcinterface.h

```diff
@@ -451,6 +464,23 @@ class IGCHeap {
     // Walks an object, invoking a callback on each member.
     virtual void WalkObject(Object* obj, walk_fn fn, void* context) = 0;
 
+    // Walk the heap object by object.
+    virtual void WalkHeap(walk_fn fn, void* context, int gen_number, BOOL walk_large_object_heap_p) = 0;
+    
+    // Walks the survivors and get the relocation information if objects have moved.
+    virtual void WalkSurvivors(void* gc_context, record_surv_fn fn, size_t diag_context, walk_surv_type type) = 0;
+
+    // Walks the finalization queue.
+    virtual void WalkFinalizeQueue(void* gc_context, fq_walk_fn fn) = 0;
+
+    // Scan roots on finalizer queue. This is a generic function.
+    virtual void ScanFinalizeQueue(fq_scan_fn fn, void* context) = 0;
+
+    // Scan handles for profiling or ETW.
+    virtual void ScanHandlesForProfilerAndETW(handle_scan_fn fn, int gen_number, ScanContext* context) = 0;
```

Comment:
It would be nice to not have `ProfilerAndETW` in the names. Call it just `ScanHandles` or `DiagScanHandles`?

Maybe have `Diag` prefix on all the diagnostic methods - to match the other direction?